tests/jax/test_custom_ops_xmap.py

```diff
@@ -0,0 +1,205 @@
+# Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```

Comment:
why different file for custom_p and xmap?

Reply:
I generally try to avoid going over 300-400 lines of code per Python file. I find that it improves readability and maintainability.

In this particular case, there's the additional convenience of simply deleting the independent xmap test file when we're ready to completely remove xmap sharding from TE. Easier and safer than stripping out xmap tests from a combined file.